src/client/components/vision2/camera/shaders/world_line.frag

```diff
@@ -0,0 +1,280 @@
+precision lowp float;
+
+#define M_PI 3.1415926535897932384626433832795
+
+// Lens/projection parameters
+uniform vec2 viewSize;
+uniform float focalLength;
+uniform int projection;
+uniform vec2 centre;
+uniform vec2 k;
+
+// Line parameters
+uniform vec3 axis;
+uniform vec3 start;
+uniform vec3 end;
+
+// Style
+uniform vec4 color;
+uniform float lineWidth;
+
+varying vec2 vUv;
+
+#define RECTILINEAR_PROJECTION 1
+#define EQUIDISTANT_PROJECTION 2
+#define EQUISOLID_PROJECTION 3
+
+// TODO(trent) these should be moved into a separate GLSL file once there is a decent #include system
```

Comment:
Does one of these exist yet? :P

Reply:
Perhaps! https://threejs.org/docs/#api/en/renderers/shaders/ShaderChunk

Reply:
Never mind, that is not what I thought it was 😞